tests/cdc_successful/cdc_successful_test.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/k8ssandra/cass-operator/tests/util/kubectl"
+	shutil "github.com/k8ssandra/cass-operator/tests/util/sh"
+)
+
+var (
+	testName            = "OSS CDC flows work"
+	namespace           = "test-cdc"
+	dcName              = "dc1"
+	dcYaml              = "../testdata/test-cdc/cassandra-datacenter.yaml"
+	pulsarValues        = "../testdata/test-cdc/dev-values.yaml"
+	testUtilsDeployment = "../testdata/test-cdc/testutils-deployment.yaml"
+	ns                  = ginkgo_util.NewWrapper(testName, namespace)
+)
+
+func TestLifecycle(t *testing.T) {
+	AfterSuite(func() {
+		logPath := fmt.Sprintf("%s/aftersuite", ns.LogDir)
+		err := kubectl.DumpAllLogs(logPath).ExecV()
+		if err != nil {
+			t.Logf("Failed to dump all the logs: %v", err)
+		}
+
+		fmt.Printf("\n\tPost-run logs dumped at: %s\n\n", logPath)
+		ns.Terminate()
+		err = kustomize.Undeploy(namespace)
+		if err != nil {
+			t.Logf("Failed to undeploy cass-operator: %v", err)
+		}
+		kubectl.Delete("ns", "pulsar").OutputPanic()
+	})
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, testName)
+}
+
+var _ = Describe(testName, func() {
+	Context("when in a new cluster with CDC enabled", func() {
+		Specify("CDC feeds will become available for read", func() {
+
+			By("creating a namespace for the cass-operator")
+			err := kubectl.CreateNamespace(namespace).ExecV()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("deploy cass-operator with kustomize")
+			err = kustomize.Deploy(namespace)
+			Expect(err).ToNot(HaveOccurred())
+			ns.WaitForOperatorReady()
+
+			step := "creating a DC"
+			k := kubectl.ApplyFiles(dcYaml)
+			ns.ExecAndLog(step, k)
+
+			step = "Deploying Pulsar"
+			if err := shutil.RunV("../../bin/helm", "repo", "add", "datastax-pulsar", "https://datastax.github.io/pulsar-helm-chart"); err != nil {
+				Fail(err.Error())
+			}
+			ns.ExecAndLog(step, k)
+			if err := shutil.RunV("../../bin/helm", "repo", "update"); err != nil {
+				Fail(err.Error())
+			}
+			ns.ExecAndLog(step, k)
+			if err := shutil.RunV("../../bin/helm", "install", "--create-namespace", "-n", "pulsar", "-f", pulsarValues, "pulsar", "datastax-pulsar/pulsar"); err != nil {
+				Fail(err.Error())
+			}
+			By("Waiting for all components ready")
+			readyGetter := kubectl.Get("pods").
+				WithFlag("selector", "app=cdc-testutil").
+				WithFlag("selector", "component=proxy").
+				WithFlag("namespace", "pulsar").
+				FormatOutput("jsonpath={.items[0].status.conditions[?(@.type=='Ready')].status}")
+			if err := kubectl.WaitForOutputContains(readyGetter, "True", 1800); err != nil {
+				Fail(err.Error())
+			}
+
+			ns.WaitForDatacenterReadyWithTimeouts(dcName, 1200, 1200)
+
+			By("Creating a testutils deployment")
+			step = "Deploying TestUtils"
+			k = kubectl.ApplyFiles(testUtilsDeployment)
+			ns.ExecAndLog(step, k)
+
+			By("Confirming testutils ready")
+			readyGetter = kubectl.Get("pods").
+				WithFlag("selector", "app=cdc-testutil").
+				WithFlag("namespace", namespace).
```

Comment:
Why is the `By` unnecessary? It seems to make the tests more self-documenting.

I'll make the other two changes you suggest. It might be good to document the e2e test functions if you plan to ask people to use them.